### PR TITLE
New DeleteUserAccount API

### PR DESCRIPTION
**Pull request by master-fury**
### Implementations
**New Files**
- _/api/DeleteUserAccount_
    * Created DeleteUserAccount API to allow users to delete his user account from a domain.
    * Receives the data from DeleteUserAccountInput i.e. userId,domain,userAccount passes to /actions/UserDetailOperationsImpl
    * Returns DeleteUserAccountOutput as response.
    * Handled all errors and exceptions
- _/model/DeleteUserAccountInput_
    *  Model to send userId, domain and userAccount to DeleteUserId API. 
- _/model/DeleteUserAccountOutput_
    * Model for sending status as response

**Functions**
* _/actions/UserDetailOperationsImpl - deleteUserAccount()_
    * Created deleteUserAccount function to fetch userId, domain, userAccount and remove them.
* _/actions/UserDetailOperationsImpl -  searchAccountAndDelete()_
    * Created to search account and remove them.

**Other**
* Added logs in all above functions

**Testing**
* Did testing using postman by sending a request and receiving a response.

### CODE REVIEW

File: src/main/java/com/psipher/application/actions/UserDetailsOperationsImpl.java [hunk 1]
```diff
@@ -125,6 +125,43 @@ public String deleteUser(String userId) throws DDBException {
...
+
+    /**
+     * Search for the account and deletes the record
+     *
+     * @param userAccountDDBModels list of account present in the domain
+     * @param userAccount          user account which is going to be deleted
+     */
+    private void searchAccountAndDelete(List<UserAccountDDBModel> userAccountDDBModels, String userAccount) {
+        for (UserAccountDDBModel userAccountDDBModel : userAccountDDBModels) {
+            if (userAccountDDBModel.getId().equals((userAccount))) {
```
Comment (rahulkrishnan221) [inline src/main/java/com/psipher/application/actions/UserDetailsOperationsImpl.java:158]: Remove extra brackets (userAccount)

File: src/main/java/com/psipher/application/actions/UserDetailsOperationsImpl.java [hunk 2]
```diff
@@ -125,6 +125,43 @@ public String deleteUser(String userId) throws DDBException {
         return status;
     }
 
+    @Override
+    public String deleteUserAccount(String userId, String domain, String userAccount) throws DDBException {
+        UserDDBModel userDDBModel = viewDetails(userId);
+        List<WebsiteDDBModel> websiteDDBModels = userDDBModel.getWebsites();
+        for (WebsiteDDBModel websiteDDBModel : websiteDDBModels) {
```
Comment (rahulkrishnan221) [inline src/main/java/com/psipher/application/actions/UserDetailsOperationsImpl.java:132]: This is a redundant code which can be refractored. This code is also been used by deletedomain function. Try to move to single function

File: src/main/java/com/psipher/application/actions/UserDetailsOperationsImpl.java (hunk 1 above)
Comment (master-fury) [reply src/main/java/com/psipher/application/actions/UserDetailsOperationsImpl.java:158]: ok, done in new update

File: src/main/java/com/psipher/application/actions/UserDetailsOperationsImpl.java (hunk 2 above)
Comment (master-fury) [reply src/main/java/com/psipher/application/actions/UserDetailsOperationsImpl.java:132]: it will make code more complex

**Pull request merged**